android/java/org/chromium/chrome/browser/settings/AppearancePreferences.java

```diff
@@ -172,6 +190,10 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             SharedPreferencesManager.getInstance().writeBoolean(
                     BravePreferenceKeys.BRAVE_TAB_GROUPS_ENABLED, (boolean) newValue);
             BraveRelaunchUtils.askForRelaunch(getActivity());
+        } else if (PREF_BRAVE_ENABLE_SPEEDREADER.equals(key)) {
+            UserPrefs.get(Profile.getLastUsedRegularProfile())
+                           .setBoolean(BravePref.SPEEDREADER_PREF_ENABLED, (boolean) newValue);
+            BraveRelaunchUtils.askForRelaunch(getActivity());
```

Comment:
do we really need to relaunch to enable it?

Reply:
Yeah I've tested it, without relaunching it doesn't change.